src/components/module-list/module-list.component.jsx

```diff
@@ -0,0 +1,60 @@
+import React, { useState } from "react";
+import "./module-list.styles.css";
+import { Card } from "../module/module.component";
+import Submit from "../submit/submit.component";
+
+const { db, auth } = require("../../firebase/index.firebase");
+const axios = require("axios");
+
+const ModuleList = () => {
+  const [nusmodules, setNusmodules] = useState([]);
+
+  const addModules = async (nusmodules) => {
+    let modules = nusmodules.split("&");
+    let modulesTaken = [];
+    let mod;
+    for (mod of modules) {
+      if (mod.includes("?")) {
+        mod = mod.split("?");
+        mod = mod[1];
+      }
+      mod = mod.split("=", 1);
+      await axios
+        .get(`https://api.nusmods.com/v2/2019-2020/modules/${mod}.json`)
+        .then(function (response) {
+          if (response.status === 200) {
+            modulesTaken.push(mod[0]);
+          }
+        })
+        .catch(function (error) {
+          console.log(error);
+        });
+    }
+
+    let moduleFields = {
+      mods_taken: modulesTaken,
+    };
+
+    var user = auth().currentUser;
+    if (user != null) {
+      let username = user.displayName;
+      db.doc(`/user/${username}`).update(moduleFields);
```

Comment:
But would it delete the other fields as well?
I used update cause I'm afraid set will just overwrite other fields with a single moduleFields

Reply:
My bad, you are right. Use .update because you want to persist the other field. I thought if you used .update the old field of mods_taken will be updated not wholly changed.